CMakeLists.txt

```diff
@@ -111,14 +110,28 @@ endif()
 add_library(mimalloc SHARED ${mi_sources})
 set_target_properties(mimalloc PROPERTIES VERSION ${mi_version} NO_SONAME "YES" OUTPUT_NAME ${mi_basename} )
 target_compile_definitions(mimalloc PRIVATE ${mi_defines} MI_SHARED_LIB MI_SHARED_LIB_EXPORT)
+target_compile_definitions(mimalloc PRIVATE $<$<BOOL:${OVERRIDE}>:MI_MALLOC_OVERRIDE>)
 target_compile_options(mimalloc PRIVATE ${mi_cflags})
 target_include_directories(mimalloc PRIVATE include PUBLIC $<INSTALL_INTERFACE:${mi_install_dir}/include>)
 target_link_libraries(mimalloc PUBLIC ${mi_libraries})
 
 # static library
 add_library(mimalloc-static STATIC ${mi_sources})
-set_target_properties(mimalloc-static PROPERTIES OUTPUT_NAME ${mi_basename})
+if(WIN32)
+  # When building both static and shared libraries on Windows,
+  # a static library should use a different output name to 
+  # avoid the conflict with the import library of a shared one.
+  string(REPLACE "mimalloc" "mimalloc-static" mi_output_name ${mi_basename})
+  set_target_properties(mimalloc-static PROPERTIES OUTPUT_NAME ${mi_output_name})
+else()
+  set_target_properties(mimalloc-static PROPERTIES OUTPUT_NAME ${mi_basename})
+endif()
 target_compile_definitions(mimalloc-static PRIVATE ${mi_defines} MI_STATIC_LIB)
+if(NOT WIN32)
+  # It is only possible to override malloc on Windows when building as a DLL.
+  # (src/alloc-override.c)
+  target_compile_definitions(mimalloc-static PRIVATE $<$<BOOL:${OVERRIDE}>:MI_MALLOC_OVERRIDE>)
```

Comment:
Here, too, it might make sense to imitate the `if(OVERRIDE MATCHES "ON")` block that was edited, and why not just move that into the `else()` clause a couple lines above?